dot/parachain/backing/integration_test.go

```diff
@@ -910,3 +910,127 @@ func TestCanNotSecondMultipleCandidatesPerRelayParent(t *testing.T) {
 
 	time.Sleep(1 * time.Second)
 }
+
+// The new leaf view doesn't clobber the old view when we update active leaves.
+func TestNewLeafDoesNotClobberOld(t *testing.T) {
```

Comment:
Not really a problem, but I have a feeling that most of the mocking preparations are reused across all testa and can be grouped to one function

Reply:
Many mocked methods are the same, but not all.
Methods are being called a different number of times in every test.

So, I don't think we should make a separate function for it.